BSP/SMPS/SMPS.c

```diff
@@ -0,0 +1,202 @@
+#include "SMPS.h"
+
+#include <stm32l4xx_hal.h>
+
+
+#define PORT_SMPS               GPIOC
+//#define PIN_SMPS_ENABLE         GPIO_PIN_4
+//#define PIN_SMPS_V1             GPIO_PIN_5
+//#define PIN_SMPS_POWERGOOD      GPIO_PIN_6
+#define PIN_SMPS_SWITCH_ENABLE  GPIO_PIN_6
+
+#define PWR_GPIO_SMPS           PWR_GPIO_C
+//#define PWR_GPIO_ENABLE         PWR_GPIO_BIT_4
+#define PWR_GPIO_SWITCH_ENABLE  PWR_GPIO_BIT_6
+#define PWR_PU_REG              PUCRC
+
+#define NUCLEO_SMPS_GPIO_CLK_ENABLE()  __HAL_RCC_GPIOC_CLK_ENABLE()
+#define NUCLEO_SMPS_GPIO_CLK_DISABLE() __HAL_RCC_GPIOC_CLK_DISABLE()
```

Comment:
Beside the names of those macros, and some empty functions (I know its temporary) it's ok to me

Reply:
it's not temporary. I made the choice to copy the SMPS file from ST examples but needed to adapt because I have not exactly same layout. It thought it could be easier to maintain keeping everything.
Note that the name of the 6 functions is even in app notes from ST. But we could change everything inside